lib/algolia.ex

```diff
@@ -47,13 +39,57 @@ defmodule Algolia do
       raise MissingAPIKeyError
   end
 
+  @type client() :: Tesla.Client.t()
+  @type index() :: String.t()
+  @type request_option() :: {:headers, Tesla.Env.headers()}
+  @type result(resp) :: {:ok, resp} | {:error, any()}
```

Comment:
nit: Have I missed it, or is this type only used as `result(map)`? If it's the case, perhaps just `response :: {:ok, map()} | {:error, any()}` would be enough?

Reply:
Nah, you didn't miss it. I just had (have?) aspirations around making these types more specific for each call. I'm not sure if/when that will happen, so I could be convinced to just simplify it for now and change it later if that ends up happening.